src/app/common/grade-icon/grade-icon.component.ts

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+import { GradeService } from '../services/grade.service';
+import { Project } from 'src/app/api/models/project';
+
+@Component({
+  selector: 'grade-icon',
+  templateUrl: './grade-icon.component.html',
+  styleUrls: ['./grade-icon.component.scss'],
+})
+export class GradeIconComponent implements OnInit, OnChanges {
+  @Input() grade?: number;
```

Comment:
Parent components may not always pass in a number. They may pass in a string. In old angular.js file, they checked first if it was a string

Reply:
Fix: Yes, add the check back.